src/pages/studyView/table/FixedHeaderTable.tsx

```diff
@@ -351,37 +351,48 @@ export default class FixedHeaderTable<T> extends React.Component<
     }
 
     getAddRemoveAllButton() {
+        const noneSelected = this.props.numberOfSelectedRows === 0;
         const allSelected =
             this.props.numberOfSelectedRows === this.props.data.length;
 
-        let dataTest: string, onClick: () => void, content: string;
-        let showButton = false;
-
-        if (allSelected && this.props.removeAll) {
-            dataTest = 'fixed-header-table-remove-all';
-            onClick = this.onRemoveAll;
-            content = 'Deselect all';
-            showButton = true;
-        } else if (this.props.addAll) {
-            dataTest = 'fixed-header-table-add-all';
-            onClick = this.onAddAll;
-            content = `Select all${
+        let selectAllContent: string;
+        let showDeselectAll = false;
+        let showSelectAll = false;
+
+        if (!noneSelected && this.props.removeAll) {
+            showDeselectAll = true;
+        }
+        if (!allSelected && this.props.addAll) {
+            showSelectAll = true;
+            selectAllContent = `Select all${
                 this.props.showSelectableNumber
                     ? ` (${this._store.dataStore.sortedFilteredData.length})`
                     : ''
             }`;
-            showButton = true;
         }
 
-        if (showButton) {
+        if (showDeselectAll || showSelectAll) {
```

Comment:
Pretty sure it's impossible for this to be false. You'd have to have a study view with 0 potential charts. If that happens, frankly, I want something to blow up. Let's get rid of the `if` and the null return block.